roles/deploy_bmh/template/bmh.yml.j2

```diff
@@ -0,0 +1,41 @@
+{%- for node in baremetal_nodes if node.split('-')[0] not in [ 'crc', 'controller', 'ocp' ]  %}
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ node }}-bmc-secret
+  namespace: openstack
+type: Opaque
+stringData:
+  username: {{ baremetal_nodes[node]['username'] }}
+  password: {{ baremetal_nodes[node]['password'] }}
+
+---
+apiVersion: metal3.io/v1alpha1
+kind: BareMetalHost
+metadata:
+  name: {{ node }}
+  namespace: openstack
+labels:
+  app: openstack
+  workload: {{ node.split('-')[0] }}
+spec:
+  bmc:
+{# TODO
+   we'll have to build the URI address from scratch when
+   we have a proper redshift support in ci-framework
+#}
+    address: ipmi://{{ baremetal_nodes[node]['address'] }}:{{ baremetal_nodes[node]['port'] }}
```

Comment:
Are we assuming ipmi always? We are going to need idrac shortly.
Anyway, the ammount of things this field takes is "huge", including ssl, and other stuff. I'd treat it as anyone treats DB connection string, as a string that the user provides.